### DISCUSSION TITLE
Mach Correction Error During Rotor Hover Simulation at 6000 RPM

### DISCUSSION THREAD

**Discussion by AndreaRapi** [byuflowlab/FLOWUnsteady#186] [Q&A]
While simulating a rotor hover at 6000 RPM, I encountered the following error after 570 out of 720 steps:

`ERROR:` LoadError: Mach correction requested on Ma = 1.898085013254046 >= 1.0
[LoadError Mach correction req.txt](https://github.com/user-attachments/files/16247397/LoadError.Mach.correction.req.txt)

I attempted to examine the functions to identify the specific correction causing this issue but could not find anything related.
Could you please help me resolve this error?

Thank you,
Andrea

**Comment by EdoAlvarezR** [comment]
Hi Andrea! 

The error is exactly what it says: The flow became supersonic somewhere along the blade. The error is thrown because the blade elements use a Prandtl–Glauert compressibility correction that is valid up to a local Mach inflow of about 0.6. 

The error is intentionally thrown to kill the simulation because it typically means that your simulation has blown up, so I recommend visualizing the simulation to determine why it blew up.

**Comment by EdoAlvarezR** [comment]
You can also confirm that the simulation has blown up by taking a look at the enstrophy monitor.

**Comment by jayjiejie** [comment]
HI Andrea and Edo

Unfortunately, I ran into the similar issue of getting the error of Mach correction error:

ERROR: LoadError: Mach correction requested on Ma = 4.468115058306802 >= 1.0

I tried to visualise the flow filed and the simulation did blow up indeed. I tried to tweak the simulation parameters (including flow condition, blade geometry discretisation) without too much success. I am wondering if you have any suggestions for approaching this Mach correction error problem?

A follow up question is weather there is any other control parameter for blade discretisation apart from n (number of blade elements per blade) and r (geometric expansion of elements)?

Thank you very much in advance.

Kind regards,
Jay
